### PR TITLE
Fix when `sys.executable` is empty or null

### PR DESCRIPTION
**Pull request by bryant1410**
There are exceptional cases in which `sys.executable` is empty or null (see [the docs](https://docs.python.org/3/library/sys.html#sys.executable)). I think it's better to just not delete the image in these cases, instead of crashing.

### CODE REVIEW

**Comment by radarhere** [comment]
Just curious - did you figure this out because you are in an environment where `sys.executable` is `None`? What type of environment is that?

**Comment by bryant1410** [comment]
Yes. I'm not sure how much details I can share, but it's kind of a custom way of running Python where the final program is compiled, so there's no Python interpreter anymore (AFAIK).

What about calling `os.remove` directly?

**Comment by radarhere** [comment]
The subprocess doesn't just call `os.remove`, it also calls `time.sleep(20)`. This behaviour started in #6010, so that the current Python process doesn't hang for 20 seconds. This period is intended to give time for Preview to open the image.

However, given that #6045 dropped removing the image altogether on Unix, the current form of this PR is not unreasonable.

**Comment by radarhere** [comment]
I've created https://github.com/bryant1410/Pillow/pull/1 as a suggestion - if `sys.executable` is not available, how about falling back to `python3` instead?

**Comment by bryant1410** [comment]
> I've created [bryant1410#1](https://github.com/bryant1410/Pillow/pull/1) as a suggestion - if `sys.executable` is not available, how about falling back to `python3` instead?

Sounds good to me! It works locally.

File: src/PIL/ImageShow.py [hunk 1]
```diff
@@ -178,14 +178,16 @@ def show_file(self, path=None, **options):
...
         subprocess.call(["open", "-a", "Preview.app", path])
-        subprocess.Popen(
-            [
-                sys.executable,
-                "-c",
-                "import os, sys, time; time.sleep(20); os.remove(sys.argv[1])",
-                path,
-            ]
-        )
+        exectable = sys.executable or shutil.which("python3")
```
Comment (radarhere) [inline src/PIL/ImageShow.py:181]:
```suggestion
        exectuable = sys.executable or shutil.which("python3")
```

**Review by radarhere — APPROVED** [review]

**Comment by beaux-riel** [comment]
I’m still experiencing the issue where the temporary file is deleted before Preview can open it on Mac. Has/will this be resolved?

**Comment by radarhere** [comment]
This PR hasn't been released yet. It will be released as part of Pillow 9.3.0, due out towards the end of October.

However, it doesn't sound like you're experiencing the same problem. Are you saying that your images are taking longer than 20 seconds to open in Preview? Are they very large?

**Pull request merged**